content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,21 +8,89 @@ preview: true
+```prisma highlight=4;normal
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId  Int
+}
 
-</Admonition>
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
 
-When you enable the `referentialActions` preview flag, you must choose one of three upgrade paths. See the [Upgrading to referential actions guides](../../../../guides/upgrade-guides/upgrading-to-referential-actions) for more information on adopting the feature in your project
+If you do not specify a referential action, Prisma [uses a default](#referential-action-defaults).
 
 </TopBlock>
 
-## What are referential actions?
+## Enable the <inlinecode>referentialActions</inlinecode> Preview feature
+
+Referential actions are a Preview feature. To enable the `referentialActions` Preview feature:
 
-Referential actions are constraints that define how a referenced record is handled by the database when you run an [`update`](../../prisma-client/crud#update) or [`delete`](../../prisma-client/crud#delete) query.
+1. [Upgrade to Prisma version 2.26.0](../../../../guides/upgrade-guides/upgrading-to-latest).
+2. Adding `referentialActions` to the `previewFeatures` field of the Prisma Client [`generator`](../../../../reference/api-reference/prisma-schema-reference#generator) block in the `prisma.schema` file:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialActions"]
+}
+```
+
+3. Follow one of two [upgrade paths for referential actions](../../../../guides/upgrade-guides/upgrading-to-referential-actions).
+
+<Admonition type="alert">
+
+It is extremely important that you follow an [upgrade path for referential actions](../../../../guides/upgrade-guides/upgrading-to-referential-actions). The `referentialAction` Preview feature **removes the safety net in Prisma Client that prevents cascading deletes at runtime**. If you enable the feature _without upgrading your database_, the [old default action]() - `ON DELETE CASCADE` - becomes active. This may result in cascading deletes that you did not expect.
+
+</Admonition>
+
+## Using referential actions
+
+Referential actions are constraints that define how a referenced record is handled by the database when you update or delete records. This includes `delete`, `deleteMany`, `update`, and `updateMany`.
+
+Referential actions are defined in the [`@relation`](../../../../reference/api-reference/prisma-schema-reference#relation) attribute and represent a **foreign key constraint** in the underlying database. If you do not specify a referential action, [Prisma falls back to a default](#referential-action-defaults).
```

Comment:
```suggestion
Referential actions are defined in the [`@relation`](../../../../reference/api-reference/prisma-schema-reference#relation) attribute and map to the actions on the **foreign key constraint** in the underlying database. If you do not specify a referential action, [Prisma falls back to a default](#referential-action-defaults).
```

I think something like this would work better?
The actions map to the actions in the foreign key, not to the foreign keys themselves.

* Relation maps to foreign key
* Referential actions on the relations map to the foreign key actions.

Reply:
I think below this paragraph we could one more? We could explain that Prisma Migrate and prisma db push will update the database schema is needed to keep the foreign key actions in sync with what's defined in the Prisma Schema.

Similarly, introspection will update the relation attributes in the Prisma Schema if needed to reflect the foreign key actions defined in the database.